src/NServiceBus.Core/Licensing/LicenseManager.cs

```diff
@@ -25,14 +30,53 @@ internal void InitializeLicense(string licenseText, string licenseFilePath)
 
             if (result.HasExpired)
             {
-                if (result.License.IsTrialLicense)
+                LogExpiredLicenseError(result.License, Logger);
+            }
+            else
+            {
+                LogWarningIfLicenseIsAboutToExpire(result.License, Logger);
+            }
+
+            if (result.HasExpired && result.License.IsTrialLicense)
+            {
+                OpenTrialExtensionPage();
+            }
+        }
+
+        internal void LogExpiredLicenseError(License activeLicense, ILog logger)
+        {
+            if (activeLicense.UpgradeProtectionExpiration.HasValue)
+            {
+                logger.Error("Upgrade protection expired. Please extend your upgrade protection so that we can continue to provide you with support and new versions of the Particular Service Platform.");
+            }
+            else if (activeLicense.IsTrialLicense)
+            {
+                logger.Error("Trial license expired. Please extend your trial or purchase a license to continue using the Particular Service Platform.");
+            }
+            else
+            {
+                logger.Error("Platform license expired. Please extend your license to continue using the Particular Service Platform.");
+            }
+        }
+
+        internal void LogWarningIfLicenseIsAboutToExpire(License activeLicense, ILog logger)
+        {
+            if (activeLicense.UpgradeProtectionExpiration.HasValue)
+            {
+                if (activeLicense.UpgradeProtectionExpiration.Value.Subtract(ExpirationWarningThreshold) <= utcDateProvider().Date)
```

Comment:
note for myself: this doesn't take into account scenarios where you're running on a version valid for the upgrade protection although the upgrade protection has expired at the current date.